### PR TITLE
feat(headless): add support for deselecting breadcrumb directly from the manager

### PR DESCRIPTION
**Pull request by nrawji**

### CODE REVIEW

File: packages/headless/package.json [hunk 1]
```diff
@@ -24,11 +24,11 @@
     "test:unit": "jest --testPathIgnorePatterns=src/integration-tests",
     "test:integration": "jest --testPathPattern=src/integration-tests",
     "npm:publish": "npm publish --access public",
-    "doc": "typedoc --tsconfig tsconfig.typedoc.json"
+    "doc": "typedoc --tsconfig tsconfig.typedoc.json"K
   },
   "dependencies": {
     "@coveo/bueno": "^0.1.0",
-    "@reduxjs/toolkit": "^1.4.0",
+    "@reduxjs/toolkit": "^1.4.0",K
```
Comment (ThibodeauJF) [inline packages/headless/package.json:31]: typo here

File: packages/atomic/src/components.d.ts [hunk 2]
```diff
@@ -84,7 +84,7 @@ export namespace Components {
     interface AtomicResultTemplate {
         "conditions": ResultTemplateCondition[];
         "fieldsToInclude"?: string;
-        "getConditions": () => Promise<ResultTemplateCondition[]>;
+        "getConditions": () => Promise<any[]>;
```
Comment (olamothe) [inline packages/atomic/src/components.d.ts:87]: 🤔  Why was this modified ?

File: packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts [hunk 3]
```diff
@@ -193,6 +194,16 @@ export const buildBreadcrumbManager = (
       dispatch(deselectAllFacets());
       dispatch(executeSearch(logClearBreadcrumbs()));
     },
+
+    /**
+     * Deselects the provided value
+     * @param value a Breadcrumb Value or a Category Facet Breadcrumb
```
Comment (olamothe) [inline packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts:200]: Suggestion: Just `@param value Breadcrumb to deselect`  ?

Also, going forward, please always include somebody from documentation team when you do changes related to JSDOC, to review.

This was a request from Frederic, from last planification, if you remember.

**Review by olamothe — APPROVED** [review]

File: packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts (hunk 3 above)
Comment (jpmarceau) [reply packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts:200]: I would also encline towards
`@param value The breadcrumb to deselect`
But maybe there's something I'm not getting about the category facet breadcrumb behaviour 🤔

Comment (nrawji) [reply packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts:200]: For non-category facets, each breadcrumb contain an array of selected values, we only want one of them. For category facets we want the entire breadcrumb because we deselect the entire path to the singular current value. Not sure if that makes sense but the key point is for category facets we want the entire breadcrumb, for the other facets we want a subvalue. Maybe I should separate them into two `deselectBreadcrumb` and `deselectCategoryFacetBreadcrumb` methods?

Comment (samisayegh) [reply packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts:200]: Honestly, I find a single method much more elegant. There are two concepts in the manager atm:

- `Breadcrumb` which holds a collection of values, and
- `BreadcrumbValue` which is an "atomic" entity that can be deselected.

I think the issue comes from mixing the two. The category facet interface currently is labelled as a `Breadcrumb` when it is really a `BreadcrumbValue` and could potentially extend the `BreadcrumbValue` interface.

Ultimately, the new function only needs the object to have a `deselect` method. I think the ideal signature would be something like:
```
deselect(breadcrumbValue: BreadcrumbValue<AnyFacetValue>) {
    breadcrumbValue.deselect()
}
```

**Review by samisayegh — APPROVED** [review]

File: packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts (hunk 3 above)
Comment (jpmarceau) [reply packages/headless/src/controllers/breadcrumb-manager/headless-breadcrumb-manager.ts:200]: That would be clearer I think. If we don't change this in this PR though, here's my doc suggestion for now:
* Deselects a facet breadcrumb value or category facet breadcrumb.
* @param value The facet breadcrumb value or a category facet breadcrumb to deselect.

**Review by jpmarceau — APPROVED** [review]
